src/modules/flight_mode_manager/tasks/Utility/Gimbal.cpp

```diff
@@ -84,9 +96,7 @@ void Gimbal::acquireGimbalControlIfNeeded()
 
 void Gimbal::releaseGimbalControlIfNeeded()
 {
-	if (_have_gimbal_control) {
-		_have_gimbal_control = false;
-
+	if (gimbalHaveControl()) {
```

Comment:
Shouldn't we do this whenever we had previously tried to acquire control? It could be that when reaching here it didn't apply yet or we didn't receive the updated status back yet and then we just stay in control.